test/lease_test.go

```diff
@@ -22,6 +24,53 @@ func TestLease(t *testing.T) {
 		g.Expect(resp.ID).To(Equal(clientv3.LeaseID(ttl)))
 		g.Expect(resp.TTL).To(Equal(ttl))
 	})
+
+	t.Run("UseLease", func(t *testing.T) {
+		var ttl int64 = 2
+		t.Run("CreateWithLease", func(t *testing.T) {
+			g := NewWithT(t)
+			{
+				resp, err := client.Lease.Grant(ctx, ttl)
+
+				g.Expect(err).To(BeNil())
+				g.Expect(resp.ID).To(Equal(clientv3.LeaseID(ttl)))
+				g.Expect(resp.TTL).To(Equal(ttl))
+			}
+
+			{
+				resp, err := client.Txn(ctx).
+					If(clientv3.Compare(clientv3.ModRevision("/leaseTestKey"), "=", 0)).
+					Then(clientv3.OpPut("/leaseTestKey", "testValue", clientv3.WithLease(clientv3.LeaseID(ttl)))).
+					Commit()
+
+				g.Expect(err).To(BeNil())
+				g.Expect(resp.Succeeded).To(BeTrue())
+			}
+		})
+
+		t.Run("LeaseExistsInKey", func(t *testing.T) {
+			g := NewWithT(t)
+			{
+				resp, err := client.Get(ctx, "/leaseTestKey", clientv3.WithRange(""))
+				g.Expect(err).To(BeNil())
+				g.Expect(resp.Kvs).To(HaveLen(1))
+				g.Expect(resp.Kvs[0].Key).To(Equal([]byte("/leaseTestKey")))
+				g.Expect(resp.Kvs[0].Value).To(Equal([]byte("testValue")))
+				g.Expect(resp.Kvs[0].Lease).To(Equal(ttl))
+			}
+		})
+
+		t.Run("KeyShouldExpire", func(t *testing.T) {
+			g := NewWithT(t)
+			// timeout ttl*2 seconds, poll 100ms
+			g.Eventually(func() []*mvccpb.KeyValue {
+				resp, err := client.Get(ctx, "/leaseTestKey", clientv3.WithRange(""))
+				g.Expect(err).To(BeNil())
+				return resp.Kvs
+			}, time.Duration(ttl*2)*time.Second, "100ms", ctx).Should(BeEmpty())
```

Comment:
Instead of hardcoding, should we add this duration as another well-defined constant in https://github.com/canonical/kine/blob/master/test/util_test.go#L15-L22?

The ttl duration is fine to stay in the test,